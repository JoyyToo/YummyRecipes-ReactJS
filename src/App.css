.App {
  text-align: center;
}

.copyright {
  text-align: center;
  float: bottom;
}

