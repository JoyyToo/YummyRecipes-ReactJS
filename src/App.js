import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import Notification from 'react-notify-toast'
import './App.css';
import Navbar from './Components/Nav'
import Landing from './Components/Landing'
import Register from './Components/Auth/Register'
import Login from './Components/Auth/Login'
import ResetPassword from './Components/Auth/ResetPass'
import NewPassword from './Components/Auth/NewPassword'
import AddCategory from './Components/Categories/Addcategory'
import Categories from './Components/Categories/Viewcategories'
import SingleCategory from './Components/Categories/SingleCategory'
import AddRecipe from './Components/Recipes/Addrecipe'
import Recipe from './Components/Recipes/Viewrecipes'
import UpdateCategory from './Components/Categories/Updatecategory'
import UpdateRecipe from './Components/Recipes/Updaterecipe'


class App extends Component {
  render() {
    return (
      <BrowserRouter>
      <div>
        <Navbar />
        <Notification />
        <Switch>
          <Route exact path="/" component={Landing} />
          <Route exact path="/register" component={Register} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/resetpassword" component={ResetPassword} />
          <Route exact path="/newpassword/:token" component={NewPassword} />
          <Route exact path="/categories" component={Categories} />
          <Route exact path="/categories/:id" component={SingleCategory} />
          <Route exact path="/addcategory" component={AddCategory} />
          <Route exact path="/updatecategory/:id" component={UpdateCategory} />
          <Route exact path="/recipes" component={Recipe} />
          <Route exact path="/addrecipe" component={AddRecipe} />
          <Route exact path="/updaterecipe" component={UpdateRecipe} />
        </Switch>
      </div>
      </BrowserRouter>
    );
  }
}

export default App;
