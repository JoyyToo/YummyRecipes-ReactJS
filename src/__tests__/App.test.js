import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';

import App from '../App';

describe('App component', () => {
    const wrapper = shallow(<App />);
  
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<App />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

    it('has the the expected components', () => {
        expect(wrapper.find('BrowserRouter')).toHaveLength(1);
        expect(wrapper.find('div').length).toBe(1);
    });

    it('renders the routes', () => {
        expect(wrapper.find('Route').length).toEqual(12);
    });

    it('renders the switch', () => {
        expect(wrapper.find('Switch')).toHaveLength(1);
    });

})