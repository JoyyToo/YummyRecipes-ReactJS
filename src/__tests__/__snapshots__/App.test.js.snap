// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders properly 1`] = `
<BrowserRouter>
  <div>
    <Nav />
    <_class
      options={Object {}}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/newpassword/:token"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/categories"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/categories/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/addcategory"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/updatecategory/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/categories/:category_id/recipes"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/categories/:category_id/addrecipe"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/categories/:category_id/recipes/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/categories/:category_id/recipe/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
